lib/code_fund/application.ex

```diff
@@ -6,15 +6,32 @@ defmodule CodeFund.Application do
   def start(_type, _args) do
     import Supervisor.Spec
 
+    options = %{keepalive: 10000, name: :slack_bot}
+    slack_token = Application.get_env(:code_fund, CodeFundWeb.Endpoint)[:slack_token]
     # Define workers and child supervisors to be supervised
-    children = [
-      # Start the Ecto repository
-      supervisor(CodeFund.Repo, []),
-      # Start the endpoint when the application starts
-      supervisor(CodeFundWeb.Endpoint, [])
-      # Start your own worker by calling: CodeFund.Worker.start_link(arg1, arg2, arg3)
-      # worker(CodeFund.Worker, [arg1, arg2, arg3]),
-    ]
+    children =
+      [
+        # Start the Ecto repository
+        supervisor(CodeFund.Repo, []),
+        # Start the endpoint when the application starts
+        supervisor(CodeFundWeb.Endpoint, [])
+        # Start your own worker by calling: CodeFund.Worker.start_link(arg1, arg2, arg3)
+        # worker(CodeFund.Worker, [arg1, arg2, arg3]),
+        # worker(Slack.Bot, [CodeFundWeb.SlackBot, [], slack_token, options], restart: :transient)
+      ] ++
+        cond do
```

Comment:
I'd prefer to not have logic or application specific config (lines 9-10)  in the application.ex file if at all possible. I understand what you're trying to achieve though -- it feels like we can wrap this logic up in a small genserver implementation that starts the Slack.Bot worker if the token is set and the add our new genserver worker here instead. Happy to help or explain more if necessary.